pkg/proc/linutil/regs_riscv64_arch.go

```diff
@@ -0,0 +1,386 @@
+		case regnum.RISCV64_S3:
+			return uint64(r.Regs.S3), nil
+		case regnum.RISCV64_S4:
+			return uint64(r.Regs.S4), nil
+		case regnum.RISCV64_S5:
+			return uint64(r.Regs.S5), nil
+		case regnum.RISCV64_S6:
+			return uint64(r.Regs.S6), nil
+		case regnum.RISCV64_S7:
+			return uint64(r.Regs.S7), nil
+		case regnum.RISCV64_S8:
+			return uint64(r.Regs.S8), nil
+		case regnum.RISCV64_S9:
+			return uint64(r.Regs.S9), nil
+		case regnum.RISCV64_S10:
+			return uint64(r.Regs.S10), nil
+		case regnum.RISCV64_S11:
+			return uint64(r.Regs.S11), nil
+		case regnum.RISCV64_T3:
+			return uint64(r.Regs.T3), nil
+		case regnum.RISCV64_T4:
+			return uint64(r.Regs.T4), nil
+		case regnum.RISCV64_T5:
+			return uint64(r.Regs.T5), nil
+		case regnum.RISCV64_T6:
+			return uint64(r.Regs.T6), nil
+		}
+	}
+
+	return 0, proc.ErrUnknownRegister
+}
+
+func (r *RISCV64Registers) SetReg(regNum uint64, reg *op.DwarfRegister) (fpchanged bool, err error) {
+	var p *uint64
+	switch regNum {
+	case regnum.RISCV64_LR:
+		p = &r.Regs.Ra
+	case regnum.RISCV64_SP:
+		p = &r.Regs.Sp
+	case regnum.RISCV64_GP:
+		p = &r.Regs.Gp
+	case regnum.RISCV64_TP:
+		p = &r.Regs.Tp
+	case regnum.RISCV64_T0:
+		p = &r.Regs.T0
+	case regnum.RISCV64_T1:
+		p = &r.Regs.T1
+	case regnum.RISCV64_T2:
+		p = &r.Regs.T2
+	case regnum.RISCV64_S0:
+		p = &r.Regs.S0
+	case regnum.RISCV64_S1:
+		p = &r.Regs.S1
+	case regnum.RISCV64_A0:
+		p = &r.Regs.A0
+	case regnum.RISCV64_A1:
+		p = &r.Regs.A1
+	case regnum.RISCV64_A2:
+		p = &r.Regs.A2
+	case regnum.RISCV64_A3:
+		p = &r.Regs.A3
+	case regnum.RISCV64_A4:
+		p = &r.Regs.A4
+	case regnum.RISCV64_A5:
+		p = &r.Regs.A5
+	case regnum.RISCV64_A6:
+		p = &r.Regs.A6
+	case regnum.RISCV64_A7:
+		p = &r.Regs.A7
+	case regnum.RISCV64_S2:
+		p = &r.Regs.S2
+	case regnum.RISCV64_S3:
+		p = &r.Regs.S3
+	case regnum.RISCV64_S4:
+		p = &r.Regs.S4
+	case regnum.RISCV64_S5:
+		p = &r.Regs.S5
+	case regnum.RISCV64_S6:
+		p = &r.Regs.S6
+	case regnum.RISCV64_S7:
+		p = &r.Regs.S7
+	case regnum.RISCV64_S8:
+		p = &r.Regs.S8
+	case regnum.RISCV64_S9:
+		p = &r.Regs.S9
+	case regnum.RISCV64_S10:
+		p = &r.Regs.S10
+	case regnum.RISCV64_S11:
+		p = &r.Regs.S11
+	case regnum.RISCV64_T3:
+		p = &r.Regs.T3
+	case regnum.RISCV64_T4:
+		p = &r.Regs.T4
+	case regnum.RISCV64_T5:
+		p = &r.Regs.T5
+	case regnum.RISCV64_T6:
+		p = &r.Regs.T6
+	case regnum.RISCV64_PC:
+		p = &r.Regs.Pc
+	}
+
+	if p != nil {
+		*p = reg.Uint64Val
+		return false, nil
+	}
+
+	switch {
+	case regNum >= regnum.RISCV64_F0 && regNum <= regnum.RISCV64_F31:
+		if r.loadFpRegs != nil {
+			err := r.loadFpRegs(r)
+			r.loadFpRegs = nil
+			if err != nil {
+				return false, err
+			}
+		}
+
+		i := regNum - regnum.RISCV64_F0
+		reg.FillBytes()
+		copy(r.Fpregset[8*i:], reg.Bytes)
+		return true, nil
+
+	default:
+		return false, fmt.Errorf("changing register %d not implemented", regNum)
+	}
+}
+
+// Refer to the definition of struct __riscv_d_ext_state in the kernel ptrace.h
```

Comment:
The comment needs to start with "RISCV64PtraceFpRegs is..."

Reply:
Fixed.